spec/lib/wolf_waagen_api/import_spec.rb

```diff
@@ -0,0 +1,250 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe WolfWaagenApi::Import do
+  let(:report) do
+    Report.create!(
+      name: 'Example Report',
+      hive_id: 'HIVE_ID',
+      start_date: DateTime.new(2019, 1, 1)
+    )
+  end
+
+  let(:other_report) do
+    Report.create!(
+      name: 'Other Report',
+      hive_id: 'HIVE_ID2'
+    )
+  end
+
+  let(:wolf_sensor_type) do
+    specified_type = WolfWaagenApi::Import::SENSOR_TYPES.first
+
+    SensorType.create!(
+      property: specified_type[:property],
+      unit: specified_type[:unit]
+    )
+  end
+
+  let(:wolf_sensor) do
+    Sensor.create!(
+      sensor_type: wolf_sensor_type,
+      name: 'Wolf Sensor 1',
+      report: report,
+      address: 1
+    )
+  end
+
+  let(:wolf_sensor2) do
+    Sensor.create!(
+      sensor_type: wolf_sensor_type,
+      name: 'Wolf Sensor 2',
+      report: report,
+      address: 2
+    )
+  end
+
+  let(:wolf_sensor_reading) do
+    Sensor::Reading.create!(
+      sensor: wolf_sensor,
+      created_at: DateTime.new(2019, 1, 2),
+      calibrated_value: 100,
+      uncalibrated_value: 100
+    )
+  end
+
+  let(:wolf_sensor_reading2) do
+    Sensor::Reading.create!(
+      sensor: wolf_sensor2,
+      created_at: DateTime.new(2019, 1, 3),
+      calibrated_value: 100,
+      uncalibrated_value: 100
+    )
+  end
+
+  let(:other_sensor_type) do
+    SensorType.create!(
+      property: 'Other Sensor Type',
+      unit: ''
+    )
+  end
+
+  let(:other_type_sensor) do
+    Sensor.create!(
+      sensor_type: other_sensor_type,
+      name: 'Other Sensor',
+      report: report,
+      address: 3
+    )
+  end
+
+  let(:other_report_sensor) do
+    Sensor.create!(
+      sensor_type: wolf_sensor_type,
+      name: 'Other Report’s Wolf Sensor',
+      report: other_report,
+      address: 4
+    )
+  end
+
+  describe '#initialize' do
+    it 'validates report' do
+      expect do
```

Comment:
Use a one-liner `expect{ described_class.new(report: nil) }.to raise_error ArgumentError`